crypto/s2n_sequence.c

```diff
@@ -30,9 +30,13 @@ int s2n_increment_sequence_number(struct s2n_blob *sequence_number)
             break;
         }
 
-        /* RFC 5246 6.1: If a TLS implementation would need to wrap a sequence number, it must
-         * renegotiate instead. We don't support renegotiation. Caller needs to create a new session.
+        /* Renegotiation is an unstable feature currently used for delayed client auth. Instead of 
+         * triggering renegotiation, throw an error.
          * This condition is very unlikely. It requires 2^64 - 1 records to be sent.
```

Comment:
We don't actually control what our customers do with renegotiation. Client auth just happens to be the most common use case. How about just:
```suggestion
        /* If a sequence number would exceed the maximum value, then we need to start a new session.
         * This condition is very unlikely. It requires 2^64 - 1 records to be sent.
```

Reply:
I think it will be nice to provide reason why we are not triggering renegotiation here. At least I feel that's what the original comment was doing and it makes the connection clear between the RFC comment below. How about
```suggestion
        /* Renegotiation is an unstable feature. If a sequence number would exceed the maximum value, 
         * then we need to start a new session instead of triggering renegotiation.
         * This condition is very unlikely. It requires 2^64 - 1 records to be sent.
```

Reply:
The issues are that 1) renegotiation doesn't solve this problem. We never initiate renegotiation. Only the peer can. 2) Even if renegotiation solved this problem, it's a bad solution that we wouldn't use. If you want to mention renegotiation, it needs to be in the context of "we're never going to renegotiate for this".

Reply:
Hmm then would it make sense to just remove the RFC reference here? Because without the mention of renegotiation, the immediate question someone reading this may have is "why are we not renegotiating?". 
So without RFC reference, the comment here may just be:
```
/* If a sequence number would exceed the maximum value, then we need to start a new session.
 * This condition is very unlikely. It requires 2^64 - 1 records to be sent.
 */
```

Or do you prefer something like:
```
/* If a sequence number would exceed the maximum value, then we need to start a new session.
 * This condition is very unlikely. It requires 2^64 - 1 records to be sent.
 * 
 * While RFC5246 suggests renegotiation for this case, we do not initiate renegotiation
 * as it can only be triggered by the peer and this feature is considered unstable.
 * Instead, we always opt to start a new session when approaching sequence number limits.
 *
 *= https://www.rfc-editor.org/rfc/rfc5246#section-6.1
 *# If a TLS implementation would need to wrap a sequence number, it must
 *# renegotiate instead.
 */
```

Reply:
Synced offline and decided to go with the former :)